src/models/data/data-set.ts

```diff
@@ -550,6 +550,11 @@ export const DataSet = types.model("DataSet", {
         }
       },
 
+      /**
+       * Remove an attribute from the dataset.
+       * Note, this method is observed by a listener in data-configuration-model.
+       * @param attributeID
+       */
```

Comment:
I understand the inclination to put a comment like this here, but `DataSet` is a foundational model that is listened to by many clients in many different contexts. I don't think it makes sense to imply that it is the responsibility of the `DataSet` to keep track of how external clients are using it. Any change to the API of the `DataSet` requires careful consideration of impacts on potential clients, whether a comment like this is present or not.